test/helpers/block.ts

```diff
@@ -0,0 +1,378 @@
+        // TODO: sudo should not have treasury event
+        const allDeposits = events
+          .filter(
+            (event) =>
+              event.section == "treasury" &&
+              event.method == "Deposit" &&
+              extrinsic.method.section !== "sudo"
+          )
+          .map((event) => (event.data[0] as any).toBigInt())
+          .reduce((p, v) => p + v, 0n);
+
+        expect(
+          txFees - txBurnt,
+          `Desposit Amount Discrepancy!\n` +
+            `    Block: #${blockDetails.block.header.number.toString()}\n` +
+            `Extrinsic: ${extrinsic.method.section}.${extrinsic.method.method}\n` +
+            `     Args: \n` +
+            extrinsic.args.map((arg) => `          - ${arg.toString()}\n`).join("") +
+            `   Events: \n` +
+            events
+              .map(({ data, method, section }) => `          - ${section}.${method}:: ${data}\n`)
+              .join("") +
+            `     fees not burnt : ${(txFees - txBurnt).toString().padStart(30, " ")}\n` +
+            `       all deposits : ${allDeposits.toString().padStart(30, " ")}`
+        ).to.eq(allDeposits);
+      }
+      sumBlockFees += blockFees;
+      sumBlockBurnt += blockBurnt;
+      previousBlockHash = blockDetails.block.hash.toString();
+    }
+  );
+
+  expect(fromPreSupply.toBigInt() - toSupply.toBigInt()).to.eq(sumBlockBurnt);
+
+  // Log difference in supply, we should be equal to the burnt fees
+  // debug(
+  //   `  supply diff: ${(fromPreSupply.toBigInt() - toSupply.toBigInt())
+  //     .toString()
+  //     .padStart(30, " ")}`
+  // );
+  // debug(`  burnt fees : ${sumBlockBurnt.toString().padStart(30, " ")}`);
+  // debug(`  total fees : ${sumBlockFees.toString().padStart(30, " ")}`);
+};
+
+export const verifyLatestBlockFees = async (
+  context: DevModeContext,
+  expectedBalanceDiff: bigint = BigInt(0)
+) => {
+  const signedBlock = await context.polkadotJs({ type: "moon" }).rpc.chain.getBlock();
+  const blockNumber = Number(signedBlock.block.header.number);
+  return verifyBlockFees(context, blockNumber, blockNumber, expectedBalanceDiff);
+};
+
+export async function jumpToRound(context: DevModeContext, round: number): Promise<string | null> {
+  let lastBlockHash = "";
+  while (true) {
+    const currentRound = (
+      await context.polkadotJs({ type: "moon" }).query.parachainStaking.round()
+    ).current.toNumber();
+    if (currentRound === round) {
+      return lastBlockHash;
+    } else if (currentRound > round) {
+      return null;
+    }
+
+    lastBlockHash = (await context.createBlock()).block.hash.toString();
+  }
+}
+
+export async function jumpBlocks(context: DevModeContext, blockCount: number) {
+  while (blockCount > 0) {
+    (await context.createBlock()).block.hash.toString();
+    blockCount--;
+  }
+}
+
+export async function jumpRounds(context: DevModeContext, count: Number): Promise<string | null> {
```

Comment:
In moonwall, with maybe support for different round meaning based on which pallet is available. Just an idea

Reply:
Good idea, raised it at https://github.com/Moonsong-Labs/moonwall/issues/140 which will be done for future PR when i cover parachain staking tests